main.js

```diff
@@ -856,7 +1064,82 @@ function main() {
         let roundDown = (num - Math.floor(num)) < eps;
         if (roundUp || roundDown) {
             return Math.round(num);
+        } else {
+            return num;
+        }
+    }
+
+    /**
+     * Takes the last value off of an array if it is zero
+     * @param {Array} array Any array of integer values
+     * @returns A new array with the last value taken out if it was zero
+     */
+    function stripLastZero(array) {
+        let newArr = [];
+        for (let i = 0; i < array.length - 1; i++) {
+            newArr.push(array[i]);
+        }
+        let last = array.pop();
+        if (last) {
+            newArr.push(last);
+        }
+        return newArr;
+    }
+
+    // from https://stackoverflow.com/a/16436975
+    function arraysEqual(a, b) {
+        if (a === b) return true;
+        if (a == null || b == null) return false;
+        if (a.length !== b.length) return false;
+        for (let i = 0; i < a.length; i++) {
+            if (a[i] !== b[i]) return false;
+        }
+        return true;
+    }
+
+    /**
+     * TODO: useless helper now, go back and replace w/ THREE lib function
+     * maybe not useless? fromArray doesn't seem to work with arrays of 2 values
+     * @param {Array} array An array of number values
+     */
+    function arrayToVector3(array) {
+        let outVector = new THREE.Vector3();
+        let x = array[0];
+        let y = array[1];
+        let z = array[2];
+        if (x) outVector.x = x;
+        if (y) outVector.y = y;
+        if (z) outVector.z = z;
+        return outVector;
+    }
+
+    function degToRad(degrees) {
+        return degrees * (Math.PI / 180);
+    }
+
+    /**
+     * Given an array of arrays representing points, pads 'points' that do not have three values
+     * with zeroes on the end, removes the last element of 'points' that have four elements. Does
+     * not handle 'points' with n < 2 or n > 4 elements. This is necessary because FOLD objects
+     * sometimes do not store the third value in the point, and assume it to be zero.
+     * @param {Array} pointList A list of Arrays, representing points in 3D space
+     * @returns {Array} The same list but with every point modified to be in 3D
+     */
+    function enforce3DCoordinates(pointList) {
+        let outArray = [];
+        for (let i = 0; i < pointList.length; i++) {
+            let point = pointList[i];
+            if (point.length === 2) {
+                outArray = outArray.concat(point);
+                outArray.push(0);
+            } else if (point.length === 4) {
+                point.pop();
```

Comment:
I think you want this to be outArray.pop() and put it after the next call; while length-4 points are unlikely to come up (I think), this would modify the array passed in if it had any, and I'm pretty sure that's not what you want to do.